core/src/main/java/org/elasticsearch/common/io/Streams.java

```diff
@@ -241,4 +243,11 @@ public static void readAllLines(InputStream input, Callback<String> callback) th
             }
         }
     }
+
+    public static byte[] toByteArray(InputStream in) throws IOException {
+        try (BytesStreamOutput out = new BytesStreamOutput()) {
+            Streams.copy(in, out);
```

Comment:
Good catch, my bad. 

Add a another cp block inside of the method, basically copied from ByteStreams.copy